src/staking/manager.ts

```diff
@@ -141,6 +141,166 @@ export class BabylonBtcStakingManager {
     };
   }
 
+  /**
+   * Create a signed staking expansion transaction that is ready to be sent to
+   * the Babylon chain.
+   */
+  async stakingExpansionRegistrationBabylonTransaction(
+    stakerBtcInfo: StakerInfo,
+    stakingInput: StakingInputs,
+    babylonBtcTipHeight: number,
+    inputUTXOs: UTXO[],
+    feeRate: number,
+    babylonAddress: string,
+    // Previous staking transaction info
+    previousStakingTxInfo: {
+      stakingTx: Transaction,
+      paramVersion: number,
+      stakingInput: StakingInputs,
+    },
+  ): Promise<{
+    signedBabylonTx: Uint8Array;
+    stakingTx: Transaction;
+  }> {
+    // Perform validation for the staking expansion inputs
+    validateStakingExpansionInputs(
+      {
+        babylonBtcTipHeight,
+        inputUTXOs,
+        stakingInput,
+        previousStakingInput: previousStakingTxInfo.stakingInput,
+        babylonAddress,
+      },
+    );
+    // Param for the expandsion staking transaction
+    const params = getBabylonParamByBtcHeight(
+      babylonBtcTipHeight,
+      this.stakingParams,
+    );
+
+    const paramsForPreviousStakingTx = getBabylonParamByVersion(
```

Comment:
there slight edge case against which  Babylon node protects against but it is not reflected here i.e if more that `paramsForPreviousStakingTx.covenantQuorum` covenant members have been rotated out of `paramsForPreviousStakingTx` ,Babylon will reject stake expansion request

we could add such check here